client/src/components/RecipeShow.js

```diff
@@ -11,26 +16,140 @@ const RecipeShow = (props) => {
+        } else {
+          return false
+        }
+      })
+      setUpdatedIngredients(ingredientsToUpdate)
+    }
+    
+    const compareIngredient = (ingredient) => {
+      const ingredientInPantry = pantryIngredients.find(pantryIngredient => pantryIngredient.name.toLowerCase() === ingredient.name)
+      if(ingredientInPantry) {
+        if(ingredientInPantry.unit === ingredient.unit) {
+          if(ingredientInPantry.amount >= ingredient.amount) {
+            return "have enough"
+          } else {
+            return "need more"
+          }
+        } else {
+          const convertedRecipeIngredient = Ingredients.convertIngredient(ingredient)
+          const convertedPantryIngredient = Ingredients.convertIngredient(ingredientInPantry)
+          if (!convertedRecipeIngredient || !convertedPantryIngredient) {
+            return "can't tell"
+          } else {
+            if (convertedPantryIngredient - 3 >= convertedRecipeIngredient) {
+              return "have enough"
+            } else if (convertedPantryIngredient + 3 <= convertedRecipeIngredient) {
+              return "need more"
+            } else {
+              return "it's close"
+            }
+          }
+        }
+      } else {
+        return "don't have"
+      }
+    }
+
+    const makeRecipe = async (event) => {
+      let updatedIngredientData = updatedIngredients.filter(element => element)
+      updatedIngredientData.forEach(ingredient => {
+        ingredient.amount = Number(ingredient.amount.toFixed(2))
+        if(ingredient.amount <= 0) {
+          ingredient.amount = 0.01
+        }
+      })
+      try {
+        const response = await fetch("/api/v1/recipes/make", {
+          method: "POST",
+          headers: new Headers({
+            "Content-Type": "application/json"
+          }),
+          body: JSON.stringify(updatedIngredientData)
+        })
+        if(response.ok) {
+          fetchPantryIngredients()
+          setMade(true)
+        }
+      } catch(err) {
+        console.error(`Error in Fetch: ${err.message}`)
+      }
+    }
+
+    const ingredientsList = recipeIngredients.map(ingredient => {
+      const ingredientInPantry = compareIngredient(ingredient)
+      let amountNote
+      if(ingredientInPantry === "don't have") {
+        amountNote = <p className="red">It looks like you don't have this ingredient</p>
+      } else if(ingredientInPantry === "need more") {
+        amountNote = <p className="red">You don't have enough of this ingredient</p>
+      } else if(ingredientInPantry === "can't tell") {
+        amountNote = <p className="yellow">We can't tell if you have enough of this ingredient</p>
+      } else if(ingredientInPantry === "it's close") {
+        amountNote = <p className="yellow">You have just about enough, but you might run out of this ingredient</p>
+    } else if(ingredientInPantry === "have enough") {
+      amountNote = <p className="green">You have more than enough of this ingredient</p>
+    }
```

Comment:
PORO logic

Reply:
the <p> tag can remain here though. The text could be delegated to an IngredientReconciler PORO. Not a priority though, this logic is okay as it is.